src/plugins/intel_gpu/src/kernel_selector/cl_kernels/scatter_elements_update_ref.cl

```diff
@@ -22,6 +22,47 @@
     #error "OUTPUT_DIMS is supposed to be same as INPUT2_DIMS"
 #endif
 
+#ifdef REDUCE_MODE
+    #define SUM 1
+    #define PROD 2
+    #define MIN 3
+    #define MAX 4
+    #define MEAN 5
+
+    #if USE_INIT_VAL == 0
+        #if REDUCE_MODE == SUM
+            #define REDUCTION_NEUTRAL_VALUE INPUT0_VAL_ZERO
+        #elif REDUCE_MODE == PROD
+            #define REDUCTION_NEUTRAL_VALUE INPUT0_VAL_ONE
+        #elif REDUCE_MODE == MIN
+            #define REDUCTION_NEUTRAL_VALUE INPUT0_VAL_MAX
+        #elif REDUCE_MODE == MAX
+            #define REDUCTION_NEUTRAL_VALUE INPUT0_VAL_MIN
+        #elif REDUCE_MODE == MEAN
+            #define REDUCTION_NEUTRAL_VALUE INPUT0_VAL_ZERO
+        #else
+            #error "Invalid REDUCE_MODE value"
+        #endif
+    #endif
+
+    inline INPUT2_TYPE FUNC(reduce)(INPUT2_TYPE a, INPUT2_TYPE b)
+    {
+    #if REDUCE_MODE == SUM
+        return a + b;
+    #elif REDUCE_MODE == PROD
+        return a * b;
+    #elif REDUCE_MODE == MIN
+        return min(a, b);
+    #elif REDUCE_MODE == MAX
+        return max(a, b);
+    #elif REDUCE_MODE == MEAN
+        return (a + b) / (INPUT2_TYPE)(1 + USE_INIT_VAL);
```

Comment:
If use_init_var == false, mean is just the new updated val?

Reply:
Yes. 
It may look odd, but that's how it works in reference implementation: when use_init_val == false, MEAN op is performed between updates value and 0 (reduction neutral value for MEAN), and divisor (counter) does not include 1 for this zero - see [line 244 in scatter_elements_update.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/core/reference/include/openvino/reference/scatter_elements_update.hpp#L244)